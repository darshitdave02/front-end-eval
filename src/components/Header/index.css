@import url('https://fonts.googleapis.com/css2?family=Big+Shoulders+Inline+Display:wght@700&display=swap');

.header {
  background-color: black;
  padding-left: 25vw;
  padding-top: 15px;
  padding-bottom: 15px;
}

.title {
  font-family: 'Big Shoulders Inline Display', cursive;
  font-weight: bold;
  color: white;
  font-size: 60px;
}

@media screen and (min-width: 320px) and (max-width: 480px) {
  .header {
    padding-left: 5vw;
  }

  .title {
    font-size: 50px;
  }
}
