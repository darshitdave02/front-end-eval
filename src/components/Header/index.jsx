import React from 'react';
import { useNavigate } from 'react-router-dom';
import './index.css';

function Header() {

  const navigate = useNavigate();

  const navigateToEventDetails = () => {
    navigate('/');
  }


  return (
    <div className='header'>
      <div onClick={navigateToEventDetails} className='title'>EVENTIFY</div>
    </div>
  );
}

export default Header;
