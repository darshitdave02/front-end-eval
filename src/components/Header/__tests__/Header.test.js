import Header from '..';

import { render } from '@testing-library/react';
import { useNavigate } from 'react-router-dom';

jest.mock('react-router-dom');

describe('Header', () => {
  it('renders Header component', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should navigate to all events page upon click on header icon', () => {
    const navigate = jest.fn();
    useNavigate.mockReturnValue(navigate);
    const element = render(<Header />);
    const button = element.getByText('EVENTIFY');
    button.click();
    expect(navigate).toBeCalledWith('/');
  });
});
