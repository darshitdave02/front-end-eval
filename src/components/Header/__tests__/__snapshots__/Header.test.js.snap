// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header component 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="title"
    >
      EVENTIFY
    </div>
  </div>
</DocumentFragment>
`;
